iocage_lib/ioc_fetch.py

```diff
@@ -251,16 +253,16 @@ def fetch_release(self, _list=False):
             if os.path.isdir(dataset):
                 pass
             else:
-                self.zpool.create(pool_dataset, {"compression": "lz4"})
-                self.zfs.get_dataset(pool_dataset).mount()
+                self.zpool.create_dataset({
+                    'name': pool_dataset, 'properties': {'compression': 'lz4'}
+                })
 
             for f in self.files:
                 if not os.path.isfile(f):
 
-                    _dataset = self.zfs.get_dataset(pool_dataset)
-
-                    _dataset.umount()
-                    _dataset.delete()
+                    ds = Dataset(pool_dataset)
+                    ds.umount()
```

Comment:
I dont think you need umount here, zfs destroy does it. umount was required in py-libzfs